liquidator/liquidator.js

```diff
@@ -91,25 +91,98 @@ class Liquidator {
   queryAndWithdrawRewards = async () => {
     Logger.info({
       at: "liquidator",
-      message: "Checking for liquidations",
-      inputPrice: priceFeed
+      message: "Checking for expired and disputed liquidations to withdraw rewards from"
     });
 
     // Update the client to get the latest information.
     await this.empClient._update();
 
-    // TODO: Just showing an example of how I want to use the client:
-    // Get expired liquidations from the client.
+    let withdrawPromises = []; // store all promises to resolve in parallel later on.
+
+    // Get all expired liquidations from the client.
     const expiredLiquidations = this.empClient.getExpiredLiquidations();
-    // TODO: 
-    // - Withdraw rewards
-    // - Check that amount of rewards was correct.
+    if (expiredLiquidations.length > 0) {
+      Logger.info({
+        at: "liquidator",
+        message: "expired liquidations detected!",
+        number: expiredLiquidations.length,
+        expiredLiquidations: expiredLiquidations
+      });
+    } else {
+      Logger.info({
+        at: "liquidator",
+        message: "No expired liquidations"
+      });
+    }
+    for (const liquidation of expiredLiquidations) {
+      Logger.info({
+        at: "liquidator",
+        message: "withdrawing rewards from expired liquidation",
+        address: liquidation.sponsor,
+        id: liquidation.id
+      });
+
+      // Confirm that liquidation has eligible rewards to be withdrawn.
+      let withdraw = this.empContract.methods.withdrawLiquidation(liquidation.id, liquidation.sponsor)
+      let withdrawAmount = await withdraw.call({ from: this.account })
```

Comment:
general point/question: what happens in the case where this reverts? it's a `.call` so you are looking for the value returned but say this is called before the DVM returns a price or the liquidation was disputed successfully and so this address actually can't call this function. what is put into this var? it will fail the `if` on the following line no doubt but I'm just curious what happens with a `.call` that would revert.

Reply:
yeah you're totally right. This does revert because in `withdrawLiquidation`, `_settle()`, which calls `_getOraclePrice()`, reverts. I added a try-catch here